src/core/propagation/velocity_verlet_inline.hpp

```diff
@@ -35,7 +35,9 @@
  *  v(t) + 0.5 \Delta t f(t)/m \f] <br> \f[ p(t+\Delta t) = p(t) + \Delta t
  *  v(t+0.5 \Delta t) \f]
  */
-inline void velocity_verlet_propagate_vel_pos(const ParticleRange &particles) {
+template <typename ParticleIterable>
+inline void
+velocity_verlet_propagate_vel_pos(const ParticleIterable particles) {
```

Comment:
Here and elsewhere, is there any performance penalty in passing the `ParticleIterable` (aka `boost::iterator_range`) by copy instead of by reference?